Use gradle platform to provide a BOM

bdellegrazie opened the pull request:
Hi,

This change modifies the build process to produce a software bill of materials (BOM) as an additional artefact. We need to consume this artefact in downstream projects to simplify dependency management.

To do this, I switched to using newer constructs in Gradle, in particular the `maven` plugin has been replaced with the `maven-publish` plugin with consequent changes to the signing configuration.
Overall the configuration complexity has been reduced very slightly.

All tests pass and the publishToMavenLocal task looks correct. However I can't test the signing task, which _might_ need its conditional `Required()` check to work

Thanks!

mjpt777 commented:
The changes fail the build and are very extensive. I'm not tempted to accept this. Maybe @tmontgomery has a view.  Also a number of plugins have been removed which others require.

bdellegrazie commented:
I'll take a look at the CI output, I only ran build / test locally thus far.
Wasn't expecting anyone to take this without further discussion but please allow me to fix the build process first.

@mjpt777  Is there a particular plugin's removal you're concerned about?

mjpt777 commented:
Why are you removing plugins?

bdellegrazie commented:
@mjpt777 Switching from older `maven` plugin to `maven-publish` plugin to simplify the publication part. CI failed because I neglected to update the default build task (fixed)

tmontgomery commented:
I'm pretty sure this will fail when tried to use to push artifacts for publishing. Or if not, likely require quite a few fixes to make it pass cleanly. Unless something has changed, we can't use maven-publish.

Is there a reason for some of these changes that is required for the BOM? Or just easier? If easier, would be better not to change them at the same time as other things. So, would suggest finding the minimum needed for the BOM that touches as little as possible. Can be cleaned up separately, if needed.

bdellegrazie commented:
@tmontgomery I will rework this to use `maven` plugin as before, thanks, I wasn't aware there would be an issue publishing to maven central with the newer plugin, its a shame though as it does make the configuration and signing syntax significantly simpler.

bdellegrazie commented:
@tmontgomery: I've reverted back to old maven plugin and rebased. However at present I cannot find a way to link the old legacy publication method. I've got as far as I can get without switching to the newer publication mechanism. At this point there is nothing further I can do unless you or another member of your team can link the javaPlatform plugin to the old legacy publication mechanism

mjpt777 commented:
Closing due to lack of progress.
